lib/checking_account.rb

```diff
@@ -0,0 +1,72 @@
+require_relative 'account'
+
+module Bank
+  class CheckingAccount < Account
+
+    attr_accessor :checks_used
+
+    def initialize( id, balance, checks_used = 0)
+
+      super(id, balance)
+      @checks_used = checks_used
+
+    end #this is for initialize
+
+
+    def withdraw(amount)
+      raise ArgumentError.new if amount < 0
```

Comment:
It might be cleaner to use `super` here to call `Account#withdraw`, and centralize your logic all in one place. In this case `Account#withdraw` would probably require more parameters, like a fee and a default balance.